tests/test_smoketest.py

```diff
@@ -62,98 +69,143 @@ def setUpClass(cls):
-                      "--callback-url https://example.com/ "
-                      "--es-query '{}' "
-                      f"--replica {replica}",
-                      runner=check_output)
-            sub_id = json.loads(res.decode())["uuid"]
-            run(f"{venv_bin}hca dss get-subscriptions --replica {replica}")
-            run(f"{venv_bin}hca dss delete-subscription --replica {replica} --uuid {sub_id}")
-
-        checkout_bucket = os.environ["DSS_S3_CHECKOUT_BUCKET"]
-        bundle_id = get_upload_val(".bundle_uuid")
-        version = get_upload_val(".version")
-
+        for replica in self.replicas:
+            res = run_for_json(f'http --check {search_route} replica=={replica}',
+                               input=json.dumps({'es_query': query}).encode())
+            print(json.dumps(res, indent=4))
+            assert len(res['results']) == 1
+
+        # Wait for the checkout to complete and assert its success
+        #
         for i in range(10):
-            run(f"{venv_bin}hca dss get-bundles-checkout "
-                f"--checkout-job-id {res_checkout['checkout_job_id']} > res.json")
-            with open("res.json") as fh:
-                res = json.load(fh)
+            res = run_for_json(f"{venv_bin}hca dss get-bundles-checkout --checkout-job-id {checkout_job_id}")
             status = res['status']
             self.assertGreater(len(status), 0)
             if status == 'RUNNING':
                 time.sleep(6)
             else:
                 self.assertEqual(status, 'SUCCEEDED')
                 blob_handle = S3BlobStore.from_environment()
-                object_key = get_dst_bundle_prefix(bundle_id, version)
-                print(f"Checking bucket {checkout_bucket} object key: {object_key}")
-                files = list(blob_handle.list(checkout_bucket, object_key))
+                object_key = get_dst_bundle_prefix(bundle_uuid, bundle_version)
+                print(f"Checking bucket {self.checkout_bucket} object key: {object_key}")
+                files = list(blob_handle.list(self.checkout_bucket, object_key))
                 self.assertEqual(len(files), file_count)
                 break
         else:
             self.fail("Timed out waiting for checkout job to succeed")
 
+        # Check the notifications
+        #
+        for replica, (subscription_id, notification_key) in notifications_proofs.items():
+            obj = s3.get_object(Bucket=self.notification_bucket, Key=notification_key)
+            notification = json.load(obj['Body'])
+            self.assertEquals(subscription_id, notification['subscription_id'])
+            self.assertEquals(bundle_uuid, notification['match']['bundle_uuid'])
+            self.assertEquals(bundle_version, notification['match']['bundle_version'])
+            s3.delete_object(Bucket=self.notification_bucket, Key=notification_key)
+            run(f"{venv_bin}hca dss delete-subscription --replica {replica} --uuid {subscription_id}")
```

Comment:
If the test fails, the test subscriptions will hang around. I think it would be worth it to put this in a `finally` so that we are more sure that the subscriptions will be mopped up.

Reply:
Will do.

Reply:
I meant to say the s3 object too, but I think you got the idea.

Reply:
https://github.com/HumanCellAtlas/data-store/pull/1182/commits/dbd9d079a4ddf31572f11193734edf7f584d2f3e

which will be squashed into its parent.